paper-swipe.html

```diff
@@ -356,45 +364,73 @@
         }
       },
 
+      _directionAllowed: function (event) {
+        if (!this.noOppositeSwipe || this.swipeLeft === this.swipeRight) {
+          return true;
+        }
+
+        var towardsLeft = event.detail.dx < 0; // True when pointer is moved to the left of its initial position
+        var towardsRight = event.detail.dx > 0; // True when pointer is moved to the right of its initial position
+
+
+        if (this.swipeLeft) {
+          var beforeRightEdge = this._curPos ? event.detail.dx < -this._curPos : true;
+          return towardsLeft && !this._atEdge || (towardsRight && this._atEdge && beforeRightEdge);
+        }
+
+        if (this.swipeRight) {
+          var beforeLeftEdge = this._curPos ? event.detail.dx > -this._curPos : true;
+          return towardsRight && !this._atEdge || (towardsLeft && this._atEdge && beforeLeftEdge);
+        }
+      },
+
       _tapHandler: function (event) {
-        // when disableSwipe is true, only click event can be triggered!
-        this.fire('tap-underlay', {nodeName: 'underlay', target: event});
+        // When disableSwipe is true, only the click event can be triggered
+        this.fire('tap-underlay', { nodeName: 'underlay', target: event });
         if (sharedPanel) {
-          // release the panel after use.
+          // Release the panel after use
           sharedPanel = null;
         }
       },
 
       _onTrack: function (event) {
         if (sharedPanel && this !== sharedPanel) {
-          // release the panel.
+          // Release the panel
           sharedPanel = null;
           return;
         }
 
-        if (this._swipeAllowed()){
-          switch (event.detail.state) {
-            case 'start':
-              this._trackStart(event);
-              break;
-            case 'track':
-              this._trackX(event);
-              break;
-            case 'end':
-              this._trackEnd(event);
-              break;
+        var hover = event.detail.hover();
+        var isOverContent = hover && Polymer.dom(this._content).deepContains(hover);
+
+        if (this._swipeAllowed() && isOverContent) {
```

Comment:
@leodido Appreciated your effort. To avoid confusion, I suggest making this change as a separate issue. Currently this aims to introduce new property to eliminate swipe in opposite direction. WDYT?